src/main.js

```diff
@@ -59,9 +61,37 @@ const createWindow = async () => {
     workbenchVersion: pkg.version,
     userDataPath: app.getPath('userData'),
   };
-  ipcMain.on('variable-request', (event, arg) => {
+  ipcMain.on('variable-request', (event) => {
     event.reply('variable-reply', mainProcessVars);
   });
+  ipcMain.on('show-context-menu', (event, rightClickPos) => {
+    const template = [
+      {
+        label: 'Inspect Element',
+        click: () => {
+          BrowserWindow.fromWebContents(event.sender).inspectElement(rightClickPos.x, rightClickPos.y)
+        }
+      },
+    ]
+    const menu = Menu.buildFromTemplate(template);
+    menu.popup( BrowserWindow.fromWebContents(event.sender));
+  })
+  ipcMain.handle('show-open-dialog', async (event, arg) => {
+    const result = await dialog.showOpenDialog(arg);
+    return result;
+  });
+  ipcMain.handle('show-save-dialog', async (event, arg) => {
+    const result = await dialog.showSaveDialog(arg);
+    return result;
+  });
+  ipcMain.handle('is-dev-mode', async (event) => {
```

Comment:
Because we have this, we may no longer need the extra `webPreferences` when the `BrowserWindow` is instantiated. That was another "non-remote module" way to let the renderer know if we're in devmode. The root `index.js` looks for this flag.
```
additionalArguments: [
        ELECTRON_DEV_MODE ? '--dev' : 'packaged'
      ],
```
It could still be useful to pass the flag like this. Or we could add another IPC call instead. What do you think?

Reply:
(github won't let me comment on the relevant line)

Reply:
Hmmm good question, I'll take a look.

Reply:
I'm fine leaving this in. Six in one hand, half dozen in the other.